src/evaluator.ts

```diff
@@ -187,4 +178,23 @@ export default class Evaluator {
             );
         });
     }
+
+    private async fetch(endpoint: string): Promise<Response> {
+        const {tokenProvider, cidAssigner, appId} = this.configuration;
+
+        const token = await tokenProvider.getToken();
+        const cid = await cidAssigner.assignCid();
```

Comment:
The tokenProvider is strictly local? If it may be remote, this is make those calls in series with no need and might make an evaluation take up to 3 times the network latency.

Doing in parallel it would merge two of them, making the worst case take the latency time 2 times.

It affects the first parallel calls on the first time a user access a website or when the local storage is cleaned, probably one of the most important evaluations since it affects the first experience with our customer site.

Reply:
Overall, it's a good idea and we can apply the suggestion as it brings almost no extra cost. However, it brings no performance boost right now as the token is stored locally. I just left this door open for future use cases.

Reply:
Ok, agreed